cmd/ci-operator-config-mirror/main.go

```diff
@@ -51,6 +56,45 @@ func (o *options) validate() error {
 	return o.WhitelistOptions.Validate()
 }
 
+type dataWithInfoByRepo map[string][]config.DataWithInfo
+
+func (d dataWithInfoByRepo) deprecateRepoPaths(destinationOrgPath string) error {
+	var deprecatedPaths []string
+
+	if err := filepath.Walk(destinationOrgPath, func(path string, info os.FileInfo, err error) error {
+		if info.IsDir() && path != destinationOrgPath {
+			repo := filepath.Base(path)
+
+			if _, ok := d[repo]; !ok {
+				deprecatedPaths = append(deprecatedPaths, path)
+			}
+		}
+		return nil
+	}); err != nil {
+		return fmt.Errorf("Failed to walk through the destination dir: %v", err)
+	}
+
+	for _, deprecatedPath := range deprecatedPaths {
+		logrus.WithField("deprecated-path", deprecatedPath).Warn("Removing deprecated repository folder")
+		if err := os.RemoveAll(deprecatedPath); err != nil {
+			return fmt.Errorf("couldn't cleanup repository dir: %v", err)
+		}
+	}
```

Comment:
I am puzzled about this... We first build an index of configs by repo (and that index by definition only contains repos for which we have source configs), then delete all subdirectories in the destination directory that we have in that index. Why can't we simply delete all subdirectories in the destination directory? It would be simpler, and would take care of the care where the entire source repo config was removed (or e.g. removed from the whitelist).

So, for example, given this input:
```
.
├── openshift
│   └── api
│       ├── openshift-api-master.yaml
│       └── OWNERS
└── openshift-priv
    ├── api
    │   ├── openshift-priv-api-master.yaml
    │   ├── openshift-priv-api-REMOVED-FROM-SOURCE.yaml
    │   └── OWNERS
    └── stale
        ├── OWNERS
        └── openshift-priv-stale-master.yaml
```

The current code will only clean up the `REMOVED-FROM-SOURCE` config from `openshift-priv/api` but not the entire `openshift-priv/stale` directory which is IMO deprecated too and should be removed.

Reply:
We don't want to remove the whole directory, because there is always a chance that only a branch will need to be removed.

Reply:
> We don't want to remove the whole directory, because there is always a chance that only a branch will need to be removed.

Yeah, but in that case the configs for the remaining branches will be normally mirrored from the source configs.